controllers/volsync/vshandler_test.go

```diff
@@ -664,6 +664,14 @@ var _ = Describe("VolSync_Handler", func() {
 					Expect(k8sClient.Create(ctx, dummyPSKSecret)).To(Succeed())
 					Expect(dummyPSKSecret.GetName()).NotTo(BeEmpty())
 
+					// Make sure the secret is created to avoid any timing issues
+					Eventually(func() error {
+						return k8sClient.Get(ctx, types.NamespacedName{
```

Comment:
We just created it above and without errors, why wait for a `Get` to succeed? Only explanation I can think of is that a local cache is used to get the `Secret` and that is not yet populated, but here we use k8sclient which is not the APIReader client.

Do we have logs from a failure to know what failed here, and that the code failed to find the secret that caused the problem?

Reply:
@ShyamsundarR , yes, we have logs, the most recent one is here: https://github.com/RamenDR/ramen/actions/runs/6065880460/job/16455849301?pr=1048

Reply:
The issue here is the same as one pointed out [here](https://github.com/RamenDR/ramen/pull/1028#discussion_r1316088389). The client passed to the test is a caching client, and hence we are looping until the local cache is updated with the created resources. We can instead use a [non-caching client](https://github.com/RamenDR/ramen/blob/5b47b949c8931be1796f6584081d1739009b105a/controllers/suite_test.go#L194) for tests that do not need the functionality, which would have an additional benefit of speeding up the tests as well (instead of waiting for the cache to be updated for each case).

Reply:
@ShyamsundarR I changed the client to non caching client in addition to Eventually block after creating secret. Actually, now we can get rid of this block (and not only here), but in 3 other places in same test file.